package.json

```diff
@@ -46,7 +51,20 @@
         "highlight-matching-tag.styles": {
           "type": "object",
           "default": null,
-          "description": "Decorations for opening and closing tags"
+          "description": "Decorations for opening and closing tags",
+          "properties": {
```

Comment:
I did not know you could make it so detailed, I definitely see value in adding complete types here, would you like to do that?

I mean something like this:
```
"opening": {
  "type": "object",
  "properties": {
    "name": {
      "type": "object"
    },
    "left": {
      "type": "object"
    },
    "right": {
      "type": "object"
    },
    "full": {
      "type": "object"
    }
  }
},
```

TS Interfaces should help you with this:

```
interface Decoration {
  highlight?: string | vscode.ThemeColor
  underline?: string | vscode.ThemeColor
  surround?: string | vscode.ThemeColor
  custom?: vscode.DecorationRenderOptions
}

interface TagDecorations {
  name?: Decoration
  left?: Decoration
  right?: Decoration
  full?: Decoration
}

export interface TagStylerConfig {
  opening: TagDecorations
  closing?: TagDecorations
}
```

I don't mind the copy paste here if that's what's needed. Also, there's probably no need to make `custom` option detailed.

Reply:
For the most part, the configuration definitions follow the JSON schema standard. The only reason I was hesitant to make the object definition more robust is that it lacks the reusability of definitions like the standard JSON schema, so there will be duplication. I am fine with adding it, though.

Reply:
I think duplication does not matter here, as long as it brings value.